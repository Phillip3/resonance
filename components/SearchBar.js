// @flow

import React from 'react';
import { StyleSheet, View, TextInput } from 'react-native';
import { Feather } from '@expo/vector-icons';
import Colors from '../constants/Colors';
import textStyles from '../constants/TextStyles';

export default ({
  value = '',
  onChangeText = () => {
    console.log('`onChangeText` required');
  },
}: {
  value: string,
  onChangeText: Function,
}) => {
  return (
    <View style={styles.searchBar}>
      <Feather
        name={'search'}
        size={22}
        color={Colors.gray}
        style={styles.icon}
      />
      <TextInput
        style={[textStyles.body, styles.input]}
        keyboardAppearance="dark"
        returnKeyType="search"
        value={value}
        onChangeText={onChangeText}
      />
    </View>
  );
};

const searchBarBackgroundColor = 'rgba(255,255,255,0.2)';

const styles = StyleSheet.create({
  searchBar: {
    flex: 1,
    height: 40,
    position: 'relative',
    backgroundColor: searchBarBackgroundColor,
    borderRadius: 25,
  },
  icon: {
    position: 'absolute',
    left: 10,
    top: 8,
  },
  input: {
    height: 40,
    paddingLeft: 40,
    fontSize: 18,
    borderWidth: 0,
    color: Colors.white,
  },
});
