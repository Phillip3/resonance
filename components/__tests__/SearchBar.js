import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import SearchBar from '../SearchBar';

describe('<SearchBar/>', () => {
  test('render the default search bar', async () => {
    const { toJSON } = render(<SearchBar />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render the search bar with a value', async () => {
    const { toJSON } = render(<SearchBar value="Alpha" />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('change the search bar input', async () => {
    const mockChangeText = jest.fn();

    const { getByTestId } = render(<SearchBar onChangeText={mockChangeText} />);

    fireEvent.changeText(getByTestId('search-bar-input'), 'Alpha');

    expect(mockChangeText).toBeCalledWith('Alpha');
  });
});
