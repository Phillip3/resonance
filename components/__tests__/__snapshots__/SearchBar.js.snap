// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> render the default search bar 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255,255,255,0.2)",
      "borderRadius": 25,
      "flex": 1,
      "height": 40,
      "position": "relative",
    }
  }
>
  <Text />
  <TextInput
    allowFontScaling={true}
    keyboardAppearance="dark"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    returnKeyType="search"
    style={
      Object {
        "borderWidth": 0,
        "color": "#fff",
        "fontFamily": "NunitoSans-Regular",
        "fontSize": 18,
        "height": 40,
        "paddingHorizontal": 40,
      }
    }
    testID="search-bar-input"
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`<SearchBar/> render the search bar with a value 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255,255,255,0.2)",
      "borderRadius": 25,
      "flex": 1,
      "height": 40,
      "position": "relative",
    }
  }
>
  <Text />
  <TextInput
    allowFontScaling={true}
    keyboardAppearance="dark"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    returnKeyType="search"
    style={
      Object {
        "borderWidth": 0,
        "color": "#fff",
        "fontFamily": "NunitoSans-Regular",
        "fontSize": 18,
        "height": 40,
        "paddingHorizontal": 40,
      }
    }
    testID="search-bar-input"
    underlineColorAndroid="transparent"
    value="Alpha"
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 8,
        "position": "absolute",
        "right": 4,
        "top": 0,
      }
    }
  >
    <Text />
  </View>
</View>
`;
